### PR TITLE
Implementation of Distilbert and added Example of Fill Mask task for Distilbert and Roberta

### PR DESCRIPTION
**Pull request by deveshjawla**

### CODE REVIEW

File: example/DistilBert_FillMask/fill_mask.jl [hunk 1]
```diff
@@ -0,0 +1,34 @@
+using Transformers
+using FuncPipelines, TextEncodeBase
+using TextEncodeBase: nested2batch, nestedcall
+using Flux, StatsBase
+
+# conf = Transformers.HuggingFace.load_config("distilbert/distilbert-base-cased")
+# state_dict = Transformers.HuggingFace.load_state_dict("distilbert/distilbert-base-cased")
+
```
Comment (chengchingwen) [inline example/DistilBert_FillMask/fill_mask.jl:8]:
```suggestion
```

File: example/DistilBert_FillMask/fill_mask.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+tkr = Transformers.TextEncoders.BertTextEncoder(tkr) do e
+    e.process[1:5] |> Pipeline{:masked_position}(nested2batch ∘ nestedcall(isequal("[MASK]")), :token) |> e.process[6:end-1] |> PipeGet{(:token, :attention_mask, :masked_position)}()
+end
+
+model = Transformers.HuggingFace.load_model("distilbert/distilbert-base-cased", :ForCausalLM)
+
+tkr.tokenizer.tokenization
+
+tkr.tokenizer.tokenization.patterns
+
```
Comment (chengchingwen) [inline example/DistilBert_FillMask/fill_mask.jl:19]:
```suggestion
```

File: example/DistilBert_FillMask/fill_mask.jl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+query = "[MASK] is the Capital of France"
+input = Transformers.TextEncoders.encode(tkr, query)
+
+input_ids = input.masked_position
+
+Transformers.TextEncoders.decode(tkr, input.token)
+
```
Comment (chengchingwen) [inline example/DistilBert_FillMask/fill_mask.jl:26]:
```suggestion
```

File: example/Roberta_FillMask/fill_mask.jl [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+using Flux, StatsBase
+
+conf = Transformers.HuggingFace.load_config("distilbert/distilroberta-base")
+
+tkr = Transformers.HuggingFace.load_tokenizer("distilbert/distilroberta-base")
+model = Transformers.HuggingFace.hgf"distilbert/distilroberta-base:formaskedlm"
+
+tkr.tokenizer.tokenization
+
+tkr.tokenizer.tokenization.patterns
```
Comment (chengchingwen) [inline example/Roberta_FillMask/fill_mask.jl:13]:
```suggestion
model = Transformers.HuggingFace.load_model("distilbert/distilroberta-base", :ForMaskedLM)
```

File: example/Roberta_FillMask/fill_mask.jl [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+end
+
+query = "Paris is the<mask> of France."
+
+input = Transformers.TextEncoders.encode(new_tkr, query)
+
+input_ids = input.masked_position
+
+Transformers.TextEncoders.decode(new_tkr, input.token)
+
```
Comment (chengchingwen) [inline example/Roberta_FillMask/fill_mask.jl:26]:
```suggestion
```

File: example/DistilBert_FillMask/fill_mask.jl [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# conf = Transformers.HuggingFace.load_config("distilbert/distilbert-base-cased")
+# state_dict = Transformers.HuggingFace.load_state_dict("distilbert/distilbert-base-cased")
+
+tkr = Transformers.HuggingFace.load_tokenizer("distilbert/distilbert-base-cased")
+tkr = Transformers.TextEncoders.BertTextEncoder(tkr) do e
+    e.process[1:5] |> Pipeline{:masked_position}(nested2batch ∘ nestedcall(isequal("[MASK]")), :token) |> e.process[6:end-1] |> PipeGet{(:token, :attention_mask, :masked_position)}()
+end
+
+model = Transformers.HuggingFace.load_model("distilbert/distilbert-base-cased", :ForCausalLM)
```
Comment (chengchingwen) [inline example/DistilBert_FillMask/fill_mask.jl:14]: Shouldn't this be `:ForMaskedLM`?

Comment (deveshjawla) [reply example/DistilBert_FillMask/fill_mask.jl:11]: Yes Indeed.

**Comment by AbrJA** [comment]
Hi @deveshjawla I hope you are doing well,

There was an error running the tests, it's the same issue across all versions. In file [load](https://github.com/deveshjawla/Transformers.jl/blob/Before-Distilbert/src/huggingface/implementation/distilbert/load.jl) on line [286](https://github.com/deveshjawla/Transformers.jl/blob/Before-Distilbert/src/huggingface/implementation/distilbert/load.jl#L286) you have this:

```
if !isnothing(m.pooler)
        get_state_dict(HGFDistilBertModel, m.pooler.layer.dense, state_dict, joinname(prefix, "pooler.dense"))
    end
```

But the atribute `pooler` is not defined for `HGFDistilBertModel` because the line with this definition is [commented](https://github.com/deveshjawla/Transformers.jl/blob/Before-Distilbert/src/huggingface/implementation/distilbert/load.jl#L71)  

```
# pooler = DistilBertPooler(Layers.Dense(NNlib.tanh_fast, weight, bias))
```

I'm gonna try to solve it but maybe you already have the solution!

Best regards, I really appreciate your contribution

**Comment by deveshjawla** [comment]
Hi, I have fixed the error related to forcausalLM but there are failures which I don't understand. 
Could you please let me know what might be causing the following:
` ```
Load: Log Test Failed at /Users/runner/work/Transformers.jl/Transformers.jl/test/huggingface/load.jl:34
  Expression: load_model(model_name, hgf_model_name, task_type; config = cfg, cache = false)
  Log Pattern: min_level = Logging.Debug
````

**Comment by chengchingwen** [comment]
`/test/huggingface/load.jl:34` is testing if there are parameters that exist but are not found in the `state_dict` thus randomly initialized. You can set `ENV["JULIA_DEBUG"] = Transformers` before calling `load_model` in the REPL to see which parameter is being initialized.

**Comment by deveshjawla** [comment]
Hi Peter,

Thank you for your response. I used `JULIA_DEBUG=Transformers julia --project main.jl -s all distilbert/distilbert-base-cased` and the tests passed.

Screenshot 2024-12-20 at 3 10 58 PM

However, I have one issue,
An EnvironmentException at the code where the hugging face checkpoint is being loaded by the python in the following code at lines 101-118 of the file at [Transformers.jl/example/HuggingFaceValidation/main.jl](https://github.com/chengchingwen/Transformers.jl/pull/path)

```
@info "Load configure file in Python"
        global pyconfig = @tryrun begin
            cfg = hgf_trf.AutoConfig.from_pretrained(model_name, layer_norm_eps = 1e-9, layer_norm_epsilon = 1e-9)
            if cfg.model_type == "clip"
                if haskey(cfg, "text_config")
                    cfg.text_config.layer_norm_eps = 1e-9
                    cfg.text_config.layer_norm_epsilon = 1e-9
                end
                if haskey(cfg, "vision_config")
                    cfg.vision_config.layer_norm_eps = 1e-9
                    cfg.vision_config.layer_norm_epsilon = 1e-9
                end
            end
            cfg
        end "Failed to load configure file in Python, probably unsupported"
```

So if I move the above code outside the `try` statement then I have no issues. Could you please help me figure out why that is?
